src/manager.cpp

```diff
@@ -271,17 +271,54 @@ int Manager::updateKeyword(const types::Path i_vpdPath,
         return l_bytesUpdatedOnHardware;
     }
 
-    // Update keyword's value on inventory object path if present
-    if (!l_inventoryObjPath.empty())
+    try
     {
-        int l_bytesUpdatedOnDbus = updateKeywordOnDbus(
-            l_inventoryObjPath, l_fruPath, i_paramsToWriteData);
-
-        if (l_bytesUpdatedOnDbus == -1)
+        // If inventory D-bus object path is present, perform update
+        if (!l_inventoryObjPath.empty())
         {
-            return l_bytesUpdatedOnDbus;
+            types::Record l_recordName;
+            std::string l_interfaceName;
+            std::string l_propertyName;
+            types::DbusVariantType l_keywordValue;
+
+            if (const types::IpzData* l_ipzData =
+                    std::get_if<types::IpzData>(&i_paramsToWriteData))
+            {
+                l_recordName = std::get<0>(*l_ipzData);
+                l_interfaceName = constants::ipzVpdInf + l_recordName;
+                l_propertyName = std::get<1>(*l_ipzData);
+
+                // Read the keyword's value from hardware to write the same on
+                // D-bus.
+                l_keywordValue =
+                    readKeyword(l_fruPath, types::ReadVpdParams(std::make_tuple(
+                                               l_recordName, l_propertyName)));
+            }
+            else
+            {
+                throw std::runtime_error(
```

Comment:
why throw here and not a return -1?

Reply:
throwing here so that we will have the reason for failure while catching the exception down. and post which we return -1 to the caller.